lib/logstash/environment.rb

```diff
@@ -73,14 +73,6 @@ def ruby_bin
       ENV["USE_RUBY"] == "1" ? "ruby" : File.join("vendor", "jruby", "bin", "jruby")
     end
 
-    def jruby?
-      @jruby ||= !!(RUBY_PLATFORM == "java")
-    end
-
-    def windows?
-      ::Gem.win_platform?
-    end
-
```

Comment:
I think this could break some code in plugins or tests.
I am like you I hate duplication between theses files `lib/logstash/environment` and `lib/bootstrap/environment`.
This is a bit confusing when theses files are actually loaded or available depending on the current context.

I would revert this change and leave it in both files and we could create an issues to refactor this?